partiql-planner/src/test/kotlin/org/partiql/planner/internal/typer/operator/OpConcatTest.kt

```diff
@@ -31,9 +32,20 @@ class OpConcatTest : PartiQLTyperTestBase() {
             successArgs.forEach { args: List<StaticType> ->
                 val arg0 = args.first()
                 val arg1 = args[1]
-                if (args.contains(StaticType.NULL)) {
-                    (this[TestResult.Success(StaticType.NULL)] ?: setOf(args)).let {
-                        put(TestResult.Success(StaticType.NULL), it + setOf(args))
+                // TODO: Since nullable types are modeled as unions of types (where each has their own applicable casts),
+                //  concatenating SYMBOL and NULL will resolve to CONCAT(CAST(<sym> AS STRING), CAST(NULL AS STRING))
+                //  due to type precedence rules.
+                if (args.contains(StaticType.NULL) && args.contains(StaticType.SYMBOL)) {
+                    return@forEach
+                }
+                if (args.all { it is NullType }) {
+                    val nullableHighestPrecedenceType = StaticType.unionOf(StaticType.STRING, StaticType.NULL)
+                    (this[TestResult.Success(nullableHighestPrecedenceType)] ?: setOf(args)).let {
+                        put(TestResult.Success(nullableHighestPrecedenceType), it + setOf(args))
+                    }
+                } else if (args.contains(StaticType.NULL)) {
+                    (this[TestResult.Success(StaticType.unionOf(args.toSet()))] ?: setOf(args)).let {
+                        put(TestResult.Success(StaticType.unionOf(args.toSet())), it + setOf(args))
```

Comment:
I had to update a lot of tests in V1. Could you please pull those to main with this latest update. My general ask here (sorry if it's a pain) is that V1 work has addressed some fn typing bugs and you've perhaps re-implemented the fix into 0.14. Hopefully we can port the specific parts of V1 to here.

https://github.com/partiql/partiql-lang-kotlin/commit/324926dac95a4946e5c81d59f4c77a5ed5abf482#diff-6aef277927cb088c80994b061cba107d0485ae8acfc82134225788841aa8a852